server/src/main/java/org/spine3/server/aggregate/AggregatePartRepository.java

```diff
@@ -22,18 +22,32 @@
 
 import org.spine3.server.BoundedContext;
 
+import java.lang.reflect.Constructor;
+
+import static org.spine3.server.aggregate.Aggregates.createAggregatePart;
+import static org.spine3.server.aggregate.Aggregates.createAggregateRoot;
+
 /**
  * Common abstract base for repositories that manage {@code AggregatePart}s.
  *
  * @author Alexander Yevsyukov
  */
-public abstract class AggregatePartRepository<I, A extends AggregatePart<I, ?, ?>> extends AggregateRepository<I, A> {
+public abstract class AggregatePartRepository
+        <I, A extends AggregatePart<I, ?, ?>, R extends AggregateRoot<I>>
+        extends AggregateRepository<I, A> {
+
+    /**
+     * The {@code AggregateRoot} class, which instance will be injected
+     * into the {@code AggregatePart} instance during initialization.
+     */
+    private final Class<R> rootClass;
 
     /**
      * {@inheritDoc}
      */
-    protected AggregatePartRepository(BoundedContext boundedContext) {
+    protected AggregatePartRepository(BoundedContext boundedContext, Class<R> rootClass) {
```

Comment:
See my earlier comment. We probably can get the class of a root from the generic params of the part class.